include/seqan3/argument_parser/detail/format_ctd.hpp

```diff
@@ -0,0 +1,484 @@
+    build_reference_name(std::string const & app_name,
+                         unsigned argument_numeric_id)
+    {
+        return app_name + '.' + build_argument_name(argument_numeric_id);
+    }
+    
+    /*!\brief Append the XML 'declaration' node to the current CTD stream buffer.
+     *
+     * \details 
+     * The 'version' and 'encoding' attributes required by every XML declaration node are
+     * hard coded to '1.0' and 'UTF-8'.
+     */
+    void append_ctd_declaration_node() 
+    {
+        ctd_stream << "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n";
+    }
+
+    /*!\brief Append a CTD 'description' node to the current CTD stream buffer.
+     *
+     * \param[in] meta A structure storing application meta-data.
+     */
+    void append_ctd_description_node(argument_parser_meta_data const & meta) 
+    {
+        ctd_stream << "\t<description>\n";
+        if (!meta.short_description.empty())
+        {
+            ctd_stream << "\t\t" << escape_special_xml_chars(meta.short_description) << "\n";
+        }
+        ctd_stream << "\t</description>\n";
+    }
+
+    /*!\brief Append a CTD 'manual' node to the current CTD stream buffer.
+     *
+     * \param[in] meta A structure storing application meta-data.
+     */
+    void append_ctd_manual_node(argument_parser_meta_data const & meta) 
+    {
+        std::string description = {};
+
+        // Merge description lines into a single string.
+        for (auto const & l : meta.description)
+        {
+            description.append(l);
+        }
+
+        ctd_stream << "\t<manual>\n";
+        if (!meta.short_description.empty())
+        {
+            ctd_stream << "\t\t" << escape_special_xml_chars(description) << "\n";
+        }
+        ctd_stream << "\t</manual>\n";
+    }
+
+    /*!\brief Append the CTD 'clielement' subtree to the current CTD stream buffer.
+     *
+     * \param[in] prefixed_option_name The option name with one or multiple dash prepended.
+     * \param[in] reference_option_name The option name or argument id with the application name prepended.
+     */
+    void append_ctd_clielement_node(std::string const & prefixed_option_name, 
+                                    std::string const & reference_option_name)
+    {
+        ctd_stream << "\t\t<clielement "
+                   << "optionIdentifier=\"" << prefixed_option_name << "\" "
+                   << "isList=\"false\""
+                   << ">\n";
+        ctd_stream << "\t\t\t<mapping "
+                   << "referenceName=\"" <<reference_option_name << "\""
+                   << "/>\n";
+        ctd_stream << "\t\t</clielement>\n";
+    }
+
+    /*!\brief Append the CTD 'cli' subtree to the current CTD stream buffer.
+     *
+     * \param[in] meta A structure storing application meta-data.
+     */
+    void append_ctd_cli_node(argument_parser_meta_data const & meta) 
+    {
+        ctd_stream << "\t<cli>\n";
+        for (auto & f : clielement_option_callbacks)
+        {
```

Comment:
remove braces and separate for loops with a new line

Reply:
Ok, no problems !